### PR TITLE
Glimpse2 splitreference: Change python test to nf-test

### PR DESCRIPTION
**Pull request by LouisLeNezet**
## PR checklist

Glimpse2 splitreference: Change python test to nf-test

- [x] This comment contains a description of changes (with reason).
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/glimpse2/splitreference/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    test("Should run without map") {
+        config "modules/nf-core/glimpse2/splitreference/tests/nextflow.config"
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'ref1000GP', single_end:false ], // meta map
+                    file("https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/glimpse/1000GP.chr21.noNA12878.s.bcf", checkIfExists: true),
+                    file("https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/glimpse/1000GP.chr21.noNA12878.s.bcf.csi", checkIfExists: true),
```
Comment (maxulysse) [inline modules/nf-core/glimpse2/splitreference/tests/main.nf.test:20]:
```suggestion
                    file(params.modules_testdata_base_path + "delete_me/glimpse/1000GP.chr21.noNA12878.s.bcf", checkIfExists: true),
                    file("params.modules_testdata_base_path + "delete_me/glimpse/1000GP.chr21.noNA12878.s.bcf.csi", checkIfExists: true),
```

File: modules/nf-core/glimpse2/splitreference/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                    "chr21:16600000-16800000",
+                    "chr21:16600000-16800000"
+                ]
+                input[1]= [[ id:'map'],file("https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/glimpse/chr21.b38.gmap.gz", checkIfExists: true)]
```
Comment (maxulysse) [inline modules/nf-core/glimpse2/splitreference/tests/main.nf.test:52]:
```suggestion
                input[1]= [[ id:'map'],file(params.modules_testdata_base_path + "delete_me/glimpse/chr21.b38.gmap.gz", checkIfExists: true)]
```

**Comment by LouisLeNezet** [comment]
Thanks @maxulysse.
I'll update the other modules the same way !

File: modules/nf-core/glimpse2/splitreference/tests/nextflow.config [hunk 3]
```diff
@@ -7,5 +7,4 @@ process {
         ext.prefix = { "${meta.id}" }
     }
     publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
```
Comment (maxulysse) [inline modules/nf-core/glimpse2/splitreference/tests/nextflow.config:9]:
```suggestion
```

File: modules/nf-core/glimpse2/splitreference/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    name "Test Process GLIMPSE2_SPLITREFERENCE"
+    script "../main.nf"
+    process "GLIMPSE2_SPLITREFERENCE"
+    tag "glimpse2"
+    tag "glimpse2/splitreference"
+    tag "modules_nfcore"
+    tag "modules"
+
+    test("Should run without map") {
+        config "modules/nf-core/glimpse2/splitreference/tests/nextflow.config"
```
Comment (maxulysse) [inline modules/nf-core/glimpse2/splitreference/tests/main.nf.test:12]:
```suggestion
        config "./nextflow.config"
```

File: modules/nf-core/glimpse2/splitreference/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                { assert process.success },
+                // File has a timestamp in it and is in binary format, so we can only check the name
+                { assert file(process.out.bin_ref[0][1]).name == "ref1000GP_chr21_16600000_16800000.bin" }
+            )
+        }
+
+    }
+
+    test("Should run with map") {
+        config "modules/nf-core/glimpse2/splitreference/tests/nextflow.config"
```
Comment (maxulysse) [inline modules/nf-core/glimpse2/splitreference/tests/main.nf.test:40]:
```suggestion
        config "./nextflow.config"
```

File: modules/nf-core/glimpse2/splitreference/meta.yml [hunk 6]
```diff
@@ -36,6 +37,11 @@ input:
       type: string
       description: Target imputed region, excluding left and right buffers (e.g. chr20:1000000-2000000).
       pattern: "chrXX:leftBufferPosition-rightBufferPosition"
+  - meta2:
+      type: map
+      description: |
+        Groovy Map containing genomic map information
+        e.g. [ map:'GRCh38' ]
```
Comment (mashehu) [inline modules/nf-core/glimpse2/splitreference/meta.yml:44]:
```suggestion
        e.g. `[ map:'GRCh38' ]`
```

**Review by matthdsm — APPROVED** [review]

File: modules/nf-core/glimpse2/splitreference/tests/main.nf.test [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                // File has a timestamp in it and is in binary format, so we can only check the name
+                { assert file(process.out.bin_ref[0][1]).name == "ref1000GP_chr21_16600000_16800000.bin" }
+            )
```
Comment (mashehu) [inline modules/nf-core/glimpse2/splitreference/tests/main.nf.test:34]: version snapshot is missing

File: modules/nf-core/glimpse2/splitreference/tests/main.nf.test [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+                input[1]= [[ id:'map'], file(params.modules_testdata_base_path + "delete_me/glimpse/chr21.b38.gmap.gz", checkIfExists: true)]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                // File has a timestamp in it and is in binary format, so we can only check the name
+                { assert file(process.out.bin_ref[0][1]).name == "ref1000GP_chr21_16600000_16800000.bin" }
```
Comment (mashehu) [inline modules/nf-core/glimpse2/splitreference/tests/main.nf.test:61]: version snapshot is missing

**Review by mashehu — CHANGES_REQUESTED** [review]

**Comment by LouisLeNezet** [comment]
@mashehu: Is it good now ?

**Review by mashehu — APPROVED** [review]

**Pull request merged**